src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/SourceSpan.cs

```diff
@@ -113,6 +113,11 @@ internal readonly SourceSpan GetZeroWidthEndSpan()
         return new SourceSpan(FilePath, AbsoluteIndex + EndCharacterIndex, LineIndex, characterIndex: EndCharacterIndex, length: 0, lineCount: 0, EndCharacterIndex);
     }
 
+    internal readonly SourceSpan Slice(int startIndex, int endIndex)
```

Comment:
The second arg of a slice method should be the length, not the end index, or it breaks implicit slice support.